lucille-core/src/main/java/com/kmwllc/lucille/stage/CollapseChildrenDocuments.java

```diff
@@ -0,0 +1,42 @@
+package com.kmwllc.lucille.stage;
+
+import java.util.Iterator;
+import java.util.List;
+
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.Stage;
+import com.kmwllc.lucille.core.StageException;
+import com.typesafe.config.Config;
+
+public class CollapseChildrenDocuments extends Stage {
+
+  private List<String> fieldsToCopy;
+  private boolean dropChildren;
+  
+  public CollapseChildrenDocuments(Config config) { 
+    super(config, new StageSpec().withRequiredProperties("fieldsToCopy", "dropChildren"));
+    fieldsToCopy = config.getStringList("fieldsToCopy"); 
+    dropChildren = config.getBoolean("dropChildren"); 
+  }
+
+  @Override
+  public Iterator<Document> processDocument(Document doc) throws StageException {
+
+    if (doc.hasChildren()) {
+      // We need to iterate the children and copy their field values to the parent.
+      for (Document child : doc.getChildren()) {
+        for (String fieldToCopy : fieldsToCopy) {
+          if (child.has(fieldToCopy)) {
+            doc.addToField(fieldToCopy, child.getString(fieldToCopy));

```

Comment:
this is treating all child field values as strings, what if some are ints etc?

Reply:
we would need a plain untyped method on document that returns Object to do this field value copy in a way that worked for all types of objects.  For now, we've chosen string because it generally works with numbers and text...  other field values like dates or bytes probably won't get collapsed down properly here.

Reply:
you can use the following method on Document:
  public void setOrAdd(String name, Document other)

It takes a field name and another document. It'll add the field from the other document to the current one in a way that preserves the type.

We faced the same problem when implementing the collapsing feature on Connector -- this method, along with   public void setOrAddAll(Document other), was created to handle this situation properly.

Reply:
updated to use setOrAdd(fieldname, Document)